ykllvmwrap/src/ykllvmwrap.cc

```diff
@@ -34,12 +34,28 @@ using namespace llvm::symbolize;
 using namespace std;
 
 // The bitcode module loaded from the .llvmbc section of the currently-running
-// binary. This is shared across threads.
+// binary. This cannot be shared across threads and used concurrently without
+// acquiring a lock, and since we do want to allow parallel compilation, each
+// thread takes a copy of this module.
 ThreadSafeModule GlobalAOTMod;
 
-// Flag used to ensure that bitcode is loaded only once.
+// Flag used to ensure that GlobalAOTMod is loaded only once.
 once_flag GlobalAOTModLoaded;
 
+// The thread's copy of the GlobalAOTMod.
+//
+// A thread should never access this directly, but should instead go via
+// getThreadAOTMod() which deals with the necessary lazy initialisation.
+//
+// OPTIMISE_ME Copying GlobalAOTMod is quite expensive (cloneToNewContext()
```

Comment:
`OPTIMISE_ME` isn't really standard: `FIXME` is probably better?

Reply:
I thought we agreed on this name early on? I prefer to have a separate name so that we know where to look when we want to speed things up.

Reply:
I don't remember it. Also, `OPTIMISE` is a bit problematic because half the world spells it `OPTIMIZE`. Personally I think it's a `FIXME` but I'm not going to worry too much over it: it's not a big deal.

Reply:
@ptersilie what to you think?

Reply:
[Bevy](https://github.com/bevyengine/bevy) uses `PERF` for this kind of comments.

Reply:
That works for me too.

Reply:
And I!

Reply:
`PERF` it is.